### PR TITLE
add hub kubeconfig secret controller to replace mount secret directly

### PR DESCRIPTION
**Pull request by skeeey**

### CODE REVIEW

File: pkg/spoke/hubclientcert/secret_controller.go [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+	}
+
+	if err := os.MkdirAll(s.hubKubeconfigDir, 0700); err != nil {
+		return fmt.Errorf("unable to create dir %q : %w", s.hubKubeconfigDir, err)
+	}
+
+	// create/update configuration files from the secret
+	for key, data := range secret.Data {
+		configFilePath := path.Join(s.hubKubeconfigDir, key)
+		if err := ioutil.WriteFile(path.Clean(configFilePath), data, 0400); err != nil {
```
Comment (qiujian16) [inline pkg/spoke/hubclientcert/secret_controller.go]: should we compare the data?

File: pkg/spoke/hubclientcert/secret_controller.go [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+	recorder events.Recorder) factory.Controller {
+	s := &hubKubeconfigSecretController{
+		hubKubeconfigDir:             hubKubeconfigDir,
+		hubKubeconfigSecretNamespace: hubKubeconfigSecretNamespace,
+		hubKubeconfigSecretName:      hubKubeconfigSecretName,
+		spokeSecretLister:            spokeSecretInformer.Lister(),
+	}
+
+	return factory.New().
+		WithInformers(spokeSecretInformer.Informer()).
```
Comment (qiujian16) [inline pkg/spoke/hubclientcert/secret_controller.go]: we need a queuekey func here so we just enqueue a certain secret

File: pkg/spoke/hubclientcert/secret_controller.go (hunk 1 above)
Comment (skeeey) [reply pkg/spoke/hubclientcert/secret_controller.go]: done

File: pkg/spoke/hubclientcert/secret_controller.go (hunk 2 above)
Comment (skeeey) [reply pkg/spoke/hubclientcert/secret_controller.go]: done

**Comment by skeeey** [comment]
see https://github.com/open-cluster-management/backlog/issues/4992

**Pull request merged**